internal/testrunner/runners/system/servicedeployer/kubernetes.go

```diff
@@ -141,7 +142,17 @@ func findKubernetesDefinitions(definitionsDir string) ([]string, error) {
 func installElasticAgentInCluster() error {
 	logger.Debug("install Elastic Agent in the Kubernetes cluster")
 
-	elasticAgentManagedYaml, err := getElasticAgentYAML()
+	kibanaClient, err := kibana.NewClient()
+	if err != nil {
+		return errors.Wrap(err, "can't create Kibana client")
+	}
+
+	metadata, err := kibanaClient.InjectedMetadata()
```

Comment:
Nit. Not sure if this injected metadata is something documented or long term supported. We could have a method to get the kibana version, ever if under the hood there can be different methods in the future to discover it. But we can also wait for this future, it may never come :)
```suggestion
	version, err := kibanaClient.Version()
```

Reply:
Sounds like a good idea, more future oriented :) I will adjust the code.

Reply:
Fixed